lib/bolognese/readers/bibtex_reader.rb

```diff
@@ -86,7 +86,7 @@ def read_bibtex(string: nil, **options)
           "titles" => meta.try(:title).present? ? [{ "title" => meta.try(:title).to_s }] : [],
           "creators" => creators,
           "container" => container,
-          "publisher" => meta.try(:publisher).to_s.presence,
+          "publisher" => meta.try(:publisher).present? ? { "name" => meta.try(:publisher).to_s }.compact : nil,
```

Comment:
Hey hello :)

In `if` condition we are already checking if `meta.try(:publisher).present?`, it will execute the if block only if value is present, so we can remove the second check which is duplicate. 

Also we are using `compact` method, this method is useful to remove element which has nil values inside hash. But we are using `to_s` hence it is converting the `nil` to `" "` empty string.

We can remove the `compact` method safely from this line if we are going to keep `to_s`.

```
irb(main):008:0> { "name" => nil.to_s }.compact
=> {"name"=>""}
irb(main):009:0> {"name" => nil}.compact
=> {}
irb(main):010:0> 
```

we can rewrite this line like following,
```
meta.try(:publisher).present? ? { "name" => meta.publisher.to_s } : nil
```

One quick question to understand this better, `meta.publisher` is going to be a `Hash` or  a`String` ? I saw in `lib/bolognese/metadata.rb` file we are checking its type, hence just want to confirm my understanding here.

Please ignore  my comment if this does not sound correct in this context.
Thank you :)

Reply:
Thank you, Ashwini! Made the suggested change.

> One quick question to understand this better, meta.publisher is going to be a Hash or aString ? I saw in lib/bolognese/metadata.rb file we are checking its type, hence just want to confirm my understanding here.

meta.publisher is expected to a Hash. The readers and writers should all handle things correctly. There are some tests that initialize a Metadata object without reading or writing metadata, hence the check in the Metadata class for publisher as hash or string.